dan_layer/template_abi/src/lib.rs

```diff
@@ -27,7 +27,9 @@
 //! This library provides types and encoding that allow low-level communication between the Tari WASM runtime and the
 //! WASM modules.
 
+#[cfg(target_arch = "wasm32")]
 mod abi;
+#[cfg(target_arch = "wasm32")]
```

Comment:
I think we just need to stub out the call_engine function for non-wasm targets and then we won't need to have the cfg(wasms) on the rest of the code. But can clean this up fairly quickly once this PR is merged